File: vue-project/src/types/enums.ts
// src/types/enums.ts
export enum UserRole {
  USER = "user",
  ADMIN = "admin",
  FRIEND = "friend",
  MOD = "mod"
}

export enum ContentType {
  LESSON_PLAN = "lesson_plan",
  EXERCISE = "exercise",
  GAME = "game",
  IMAGE = "image",
  TRANSCRIPT = "transcript"
}

export enum TariffType {
  BASIC = "tariff_2",
  STANDARD = "tariff_4",
  PREMIUM = "tariff_6"
}

export enum ActionType {
  GENERATION = "generation",
  IMAGE = "image",
  POINTS_EARNED = "points_earned",
  POINTS_SPENT = "points_spent"
}

File: backend/app/core/enums.py
# app/core/enums.py
from enum import Enum

class UserRole(str, Enum):
    USER = "user"
    ADMIN = "admin"
    FRIEND = "friend"
    MOD = "mod"

class TransactionType(str, Enum):
    """Типы транзакций с баллами"""
    GENERATION = "generation"  # Списание за генерацию контента
    REWARD = "reward"  # Начисление за достижения/активность
    PURCHASE = "purchase"  # Покупка баллов
    ADMIN_ADD = "admin_add"  # Начисление администратором
    ADMIN_SUBTRACT = "admin_subtract"  # Списание администратором
    REFERRAL = "referral"  # Начисление за реферала
    DAILY_BONUS = "daily_bonus"  # Ежедневный бонус